.github/workflows/docs.yml

```diff
@@ -0,0 +1,81 @@
+name: Docs building
+
+on:
+  schedule:
+    - cron: "0 0 * * SUN"
+  workflow_dispatch:
+
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+# Default to bash
+defaults:
+  run:
+    shell: bash
+
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install deps
+        run: |
+          sudo apt-get -y install \
+            intltool \
+            libatk1.0-dev \
+            libjpeg-dev \
+            libjson-glib-dev \
+            libpugixml-dev \
+            libxml2-dev \
+            libxml2-utils \
+            python3-jsonschema \
+            xsltproc \
+            doxygen \
+            graphviz;
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+      - name: Install Python deps for Sphinx
+        run: |
+          pip install -r docs/requirements.txt
+      - name: Setup Pages
+        id: pages
+        uses: actions/configure-pages@v2
```

Comment:
I don't think this (& `Upload page artifact`) can't be in the dependant job

Reply:
I don't understand what you mean here. I have reused the workflow template for Hugo static website that I have used to update Github pages for 2 years now, this works just fine

Reply:
For `- name: Upload page artifact`, as far as I understand it, it can't be moved to another job because we upload the HTML content of the `build/docs` folder that will not be available anymore outside of this job. Then, `- name: Setup Pages` is needed before we do `- name: Upload page artifact` to be able to run it at all.

Reply:
> For `- name: Upload page artifact`, as far as I understand it, it can't be moved to another job because we upload the HTML content of the `build/docs` folder that will not be available anymore outside of this job.

Right, ok, we can exfiltrate that data via upload-artifact, but that's kinda pointless.

> Then, `- name: Setup Pages` is needed before we do `- name: Upload page artifact` to be able to run it at all.

I'm not following, but i don't see where `actions/configure-pages` writes anything to `build/docs`,
especially since it's only input that takes a path, `static_site_generator`, isn't specified.

Reply:
It seems `actions/configure-pages` prepares the metadata of the Github Page to which we will upload the artifacts. All actions for static website generators call it before uploading anything: https://github.com/actions/starter-workflows/tree/main/pages.